Implement Phoenix EVM and Protocol Upgrades

edwardmack opened the pull request:
<!-- Thanks for sending a pull request! Please check out our contribution guidelines: -->
<!-- https://github.com/hyperledger/besu/blob/master/CONTRIBUTING.md -->

## PR description
This PR implements EVM and Protocol upgrades as outlined in ECIP 1088 https://ecips.ethereumclassic.org/ECIPs/ecip-1088
## Fixed Issue(s)
<!-- Please link to fixed issue(s) here using format: fixes #<issue number> -->
<!-- Example: "fixes #2" -->

edwardmack committed:
remove code related to aztlan fork.

Signed-off-by: Edward Mack <ed@edwardmack.com>

edwardmack committed:
added genesis config option for Phoenix block

Signed-off-by: Edward Mack <ed@edwardmack.com>

edwardmack committed:
implement phoenix protocol spec definition

Implement Phoenix EVM and protocol upgrades as per ECIP-1088

Signed-off-by: Edward Mack <ed@edwardmack.com>

q9f reviewed (dismissed):
LGTM and runs smoothly on Mordor

Please also consider #28

edwardmack committed:
Merge branch 'master' into ed/etc_phoenix_update

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ChainSafe/besu?pullRequest=27) <br/>All committers have signed the CLA.

q9f committed:
p2p/config: update mordor classic discovery nodes (#29)

Signed-off-by: /raw PONG _GHMoaCXLT <58883403+q9f@users.noreply.github.com>

edwardmack committed:
Merge branch 'master' into ed/etc_phoenix_update

edwardmack committed:
Merge branch 'master' into ed/etc_phoenix_update

edwardmack committed:
Merge branch 'master' into ed/etc_phoenix_update

edwardmack committed:
Merge branch 'master' into ed/etc_phoenix_update

edwardmack committed:
remove unnecessary contractCreationProcessorBuilder

remove unnecessary contractCreationProcessorBuilder from Phoenix
ClassicProtocolSpec.

Signed-off-by: Edward Mack <ed@edwardmack.com>

edwardmack committed:
Merge branch 'master' into ed/etc_phoenix_update
